Sources/ObservableArray.swift

```diff
@@ -315,6 +316,9 @@ extension MutableObservableArray where Item: Equatable {
         subject.next(ObservableArrayEvent(change: .deletes(deletes), source: self))
         subject.next(ObservableArrayEvent(change: .inserts(inserts), source: self))
         subject.next(ObservableArrayEvent(change: .endBatchEditing, source: self))
+      } else {
+        self.array = array
+        subject.next(ObservableArrayEvent(change: .replace, source: self))
```

Comment:
Actually if you look up about it's still wrapped in the `lock.atomic.`

The entire `if performDiff { ... } else { ... }` is wrapped in the `lock.atomic.